### PR TITLE
fix: address qa auth and sharing regressions

### PR DESCRIPTION
**Pull request by engels74**
## Summary

This PR addresses the confirmed QA findings from the dogfood run while keeping the Plex console-token report scoped to source attribution rather than browser-console suppression.

## Changes

- Adds shared cron validation used by both `/admin/sync` client UI and server actions so validation messages stay aligned.
- Preserves invalid cron submissions in the field after server-side `fail(400, ...)` responses and renders the same inline `role="alert"` feedback used for client-side invalid values.
- Makes disabled cron save/initialize controls self-explanatory with visible text and `aria-describedby` wiring.
- Adds Plex OAuth regression coverage that classifies `app.plex.tv` and `plex.tv` console entries as third-party evidence and confirms Obzorarr-owned auth responses stay limited to browser-safe fields.
- Adds a source regression for server wrapped SummaryPage sharing to ensure `handleShare` opens `ShareModal` with the expected server wrapped props.

## Verification

- `bun test --env-file=.env.test tests/unit/plex-login.test.ts tests/unit/admin/ui-source-regressions.test.ts`
- `bun run check`
- `bunx biome check src/lib/cron/validation.ts src/routes/admin/sync/+page.server.ts src/routes/admin/sync/+page.svelte tests/unit/plex-login.test.ts tests/unit/admin/ui-source-regressions.test.ts`

## Browser QA Checklist

- OAuth login: confirm no Obzorarr-origin console entry contains token-like fields; Plex-owned console output should be classified as third-party.
- `/admin/sync`: type `not a cron`; verify inline error appears before submit and disabled controls reference the error.
- `/wrapped/2026`: advance to the final SummaryPage; click `Share`; confirm the modal opens, closes, and copy feedback appears.

### CODE REVIEW

File: src/lib/cron/validation.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const CRON_REQUIRED_MESSAGE = 'Cron expression is required';
+export const CRON_ALLOWED_CHARS_MESSAGE = 'Only digits, spaces, and * / - , are allowed';
+export const CRON_FIELD_COUNT_MESSAGE = 'Cron expression must have exactly 5 fields';
+
+export function validateCronExpression(expression: string): string {
+	const trimmed = expression.trim();
+	if (!trimmed) return CRON_REQUIRED_MESSAGE;
+	if (!/^[\d\s*/\-,]+$/.test(trimmed)) return CRON_ALLOWED_CHARS_MESSAGE;
```
[the remark this answers is not in the document]
Comment (engels74) [reply src/lib/cron/validation.ts]: Valid. The cron validator accepted tabs/newlines despite the spaces-only message. Fixed by tightening validation to literal spaces and adding regression coverage.

File: src/routes/admin/sync/+page.server.ts [hunk 2]
```diff
@@ -141,18 +141,25 @@ export const actions: Actions = requireAdminActions({
 		const cronExpression = formData.get('cronExpression');
 
 		if (!cronExpression || typeof cronExpression !== 'string') {
-			return fail(400, { error: 'Cron expression is required' });
+			return fail(400, {
+				error: 'Cron expression is required',
```
[the remark this answers is not in the document]
Comment (engels74) [reply src/routes/admin/sync/+page.server.ts]: Valid. The required cron message was duplicated in the schedule action. Fixed by reusing the shared constant from `$lib/cron/validation`.

File: src/routes/admin/sync/+page.server.ts [hunk 3]
```diff
@@ -141,18 +141,25 @@ export const actions: Actions = requireAdminActions({
...
+				cronError: 'Cron expression is required',
+				cronExpression: ''
+			});
 		}
 
-		const parsed = CronExpressionSchema.safeParse(cronExpression);
+		const parsed = parseCronExpression(cronExpression);
```
[the remark this answers is not in the document]
Comment (engels74) [reply src/routes/admin/sync/+page.server.ts]: Valid. The schedule action was manually validating the cron field outside Zod. Fixed by routing the payload through a Zod schema while preserving the existing `cronError`/`cronExpression` response shape.

**Comment by engels74** [comment]
augment review

File: tests/unit/plex-login.test.ts [hunk 4]
```diff
@@ -130,6 +161,72 @@ describe('sanitizeCompletedLoginResponse', () => {
...
+	});
+
+	it('keeps Obzorarr-owned completed auth responses limited to browser-safe fields', async () => {
+		const [completionSource, endpointSource, redirectSource] = await Promise.all([
+			Bun.file('src/lib/server/auth/login-completion.ts').text(),
+			Bun.file('src/routes/auth/plex/+server.ts').text(),
+			Bun.file('src/routes/auth/plex/redirect/+page.server.ts').text()
+		]);
+
+		const completedReturn = completionSource.match(/return \{\n\t\tuser: \{[\s\S]*?\n\t\};/)?.[0];
```
[the remark this answers is not in the document]
Comment (engels74) [reply tests/unit/plex-login.test.ts]: Valid. The source regex was coupled to exact return-block formatting. Fixed by removing the formatted extraction and keeping the route leak assertion with a whitespace-tolerant property check.

File: src/routes/admin/sync/+page.server.ts [hunk 5]
```diff
@@ -197,18 +212,21 @@ export const actions: Actions = requireAdminActions({
 		const expression =
 			cronExpression && typeof cronExpression === 'string' ? cronExpression : '0 0 * * *';
 
-		const parsed = CronExpressionSchema.safeParse(expression);
+		const parsed = parseCronExpression(expression);
```
[the remark this answers is not in the document]
Comment (engels74) [reply src/routes/admin/sync/+page.server.ts]: Valid. `initScheduler` was still validating the cron payload outside Zod. Fixed by routing it through `UpdateScheduleSchema.safeParse` while preserving the existing default and error response shape.

**Comment by engels74** [comment]
augment review

File: tests/unit/plex-login.test.ts [hunk 6]
```diff
@@ -130,6 +161,64 @@ describe('sanitizeCompletedLoginResponse', () => {
...
+		const browserAuthSource = `${endpointSource}\n${redirectSource}`;
+		for (const forbidden of [
+			'authToken',
+			'accessToken',
+			'services',
+			'resources',
+			'email',
+			'plexId'
+		]) {
+			expect(browserAuthSource).not.toMatch(new RegExp(String.raw`\b${forbidden}\s*:`));
```
[the remark this answers is not in the document]
Comment (engels74) [reply tests/unit/plex-login.test.ts]: Valid. The static source scan only rejected forbidden fields when they appeared as `key:`, so shorthand object properties like `{ authToken }` could pass. Fixed by expanding the forbidden-field pattern to also reject shorthand delimiters.

**Comment by engels74** [comment]
augment review

File: src/routes/admin/sync/+page.svelte [hunk 7]
```diff
@@ -138,7 +151,16 @@ function disconnectSSE() {
 }
 
 $effect(() => {
```
[the remark this answers is not in the document]
Comment (engels74) [reply src/routes/admin/sync/+page.svelte]: Valid. The cron sync effect was writing `$state` from reactive `form`/`data` inputs, which conflicts with the project Svelte 5 guidance to use `$effect` only for side effects and avoid state writes inside effects. Fixed by replacing that sync marker/effect with an assignable `$derived` cron expression and removing the now-unused `untrack` import.

**Comment by engels74** [comment]
augment review

File: src/routes/admin/sync/+page.server.ts [hunk 8]
```diff
@@ -197,18 +205,22 @@ export const actions: Actions = requireAdminActions({
 		const expression =
```
[the remark this answers is not in the document]
Comment (engels74) [reply src/routes/admin/sync/+page.server.ts]: Valid. `initScheduler` defaulted an explicitly submitted empty cron value before validation, so it could initialize with the default instead of returning the required validation error. Fixed by defaulting only when the field is missing and adding a regression test for explicit empty input.

**Comment by engels74** [comment]
augment review

File: src/routes/admin/sync/+page.svelte [hunk 9]
```diff
@@ -22,24 +23,32 @@ interface SyncProgress {
...
+	const error = actionData.cronError;
+	return submittedCronExpression(actionData) === expression && typeof error === 'string'
+		? error
+		: '';
+}
+
 let selectedBackfillYear = $state<string>('');
-let cronExpression = $state('0 0 * * *');
-const cronError = $derived(validateCron(cronExpression));
+let cronExpression = $derived(
```
[the remark this answers is not in the document]
Comment (engels74) [reply src/routes/admin/sync/+page.svelte]: Valid. The assignable `$derived` could recompute from route form/data and drop unsaved cron edits after unrelated failures or invalidation refreshes. Fixed by keeping a local cron draft for input/preset edits, using server/form values only as fallback, and clearing the draft only after a successful enhanced schedule update.

**Comment by engels74** [comment]
augment review
